src/Native/Unix/System.Native/pal_process.c

```diff
@@ -221,6 +232,34 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
         goto done;
     }
 
+#if HAVE_VFORK_SHM
+    // This platform has the shared-memory implementation of vfork(). For a one gigabyte process
+    // the expected performance gain of using vfork() rather than fork() is 99.5% merely due to
+    // avoiding page faults as the kernel does not need to set all writable pages in the parent
+    // process to copy-on-write because the child process is allowed to write to the parent process
+    // memory pages. Eliding the pipe for sending the error from execve() back to the parent is pure
+    // gravy.
```

Comment:
suggestion:

// With fork() parent and child processes have a separate address space. This is implemented by marking
// the memory as copy-on-write. When the memory is changed, it is copied to keep the address spaces
// separate.
// Using vfork() parent and child share the address space which means we avoid the overhead of setting
// up and maintaining separate address spaces.
// The thread of the parent process is blocked until the child exits or execs.